d2bs/kolbot/libs/bots/MFHelper.js

```diff
@@ -261,60 +240,29 @@ MainLoop:
 						} catch (killerror2) {
 							print(killerror2);
 						}
+					} else if (command.indexOf("cows") > -1) {
+						print("ÿc4MFHelperÿc0: Clear Cows");
 
-						delay(1000);
-
-						if (!me.inTown && !Pather.usePortal(null, player.name)) {
-							Town.goToTown();
-						}
-					} else {
-						print("Failed to use portal.");
-					}
-				} else if (command.indexOf("quit") > -1) {
-					break MainLoop;
-				} else if (command.indexOf("cows") > -1) {
-					print("ÿc4MFHelperÿc0: Clear Cows");
-
-					for (i = 0; i < 5; i += 1) {
-						if (Town.goToTown(1) && Pather.usePortal(39)) {
-							break;
+						if (me.area != 39) {
+							Town.goToTown(1);
```

Comment:
retain the retry structure, there is no guarantee that the cow portal will be near stash so this check doesn't make much sense

also this part of the code won't work properly because you would have taken the players tp, the area check would avoid this redundancy

Reply:
yeah I forget to move cows outside
the `else if cows` would be for the `if (me.area === player.area)`
if we failed to take leader tp (baal not done can't use player tp to cows), try to take the red portal as before